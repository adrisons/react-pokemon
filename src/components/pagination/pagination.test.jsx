import React from "react";
import renderer from "react-test-renderer";
import Pagination from "./pagination";

describe("WHEN: Pagination with no prev page link", () => {
  const component = renderer.create(<Pagination gotoNextPage="next-page" />);
  const tree = component.toJSON();
  it("THEN: should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  it("THEN: component should not display prev page button", () => {
    expect(tree.children.length).toEqual(1);
  });
});
describe("WHEN: Pagination with prev page link", () => {
  const component = renderer.create(
    <Pagination gotoNextPage="next-page" gotoPrevPage="prev-page" />
  );
  const tree = component.toJSON();
  it("THEN: should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  it("THEN: component should display prev page button", () => {
    expect(tree.children.length).toEqual(2);
  });
});
