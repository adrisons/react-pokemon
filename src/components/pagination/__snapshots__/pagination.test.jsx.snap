// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHEN: Pagination with no prev page link THEN: should match snapshot 1`] = `
<div
  className="buttons"
>
  <button
    onClick="next-page"
  >
    Next
  </button>
</div>
`;

exports[`WHEN: Pagination with prev page link THEN: should match snapshot 1`] = `
<div
  className="buttons"
>
  <button
    onClick="prev-page"
  >
    Previous
  </button>
  <button
    onClick="next-page"
  >
    Next
  </button>
</div>
`;
